core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -1593,6 +1594,21 @@ public void addView(View v) throws IOException {
         viewGroupMixIn.addView(v);
     }
 
+    /**
+     * Completely replaces views
+     */
+    public void setViews(Collection<View> views) throws IOException {
+        BulkChange bc = new BulkChange(this);
+        try {
+            this.views.clear();
+            for (View v : views) {
+                addView(v);
```

Comment:
this is still not thread safe.

If 2 threads are calling this you will end up with a junk result that matches neither of the requests.

This also seems to allow you to bypass the check that you can not remove the default view - there is no check that any of the provided views would be set as a default.  (I may be missing some subtlety here).